### DISCUSSION TITLE
How to mock an includes: type query

### DISCUSSION THREAD

**Discussion by quinnvaughn** [prisma/prisma#13817] [Q&A]
I'm currently mocking prisma as a singleton, and one of my resolvers uses as include statement in a findFirst and has a check for the nested field, and I'm having trouble mocking that without a ts-ignore. With a ts-ignore, I can just add it as a field in the mock and everything works, but I'd prefer not to do that. I'm wondering how to correctly mock it so I'm not getting a FieldArgs = {} error.

**Comment by austincrim** [comment]
Hey @quinnvaughn 👋,

Could you share your test setup and code?

**Reply by quinnvaughn** [reply]
It's production code so not really, but I can show a generalized example of the specific piece of code.

```
prismaMock.user.findFirst.mockResolvedValue({
id: 0,
username: 'quinn',
profile: {
// this is where the issue occurs.
id: 0,
city: 'Big City'
}
```

So in the resolver it would have something like 

```
const user = await prisma.user.findFirst({
where: ...
include: {
profile: true
}
})
```

And I'm getting an error in the mock that it's not a valid value, but the actual test works fine when I just add a // @ts-ignore 

So really just wondering how to correctly mock an include in a prisma mock (which is using the singleton method from the unit testing guide).

**Reply by thgaskell** [reply]
Strangely, TypeScript doesn't complain for me when the mocked resolved value is assigned to a variable before passing it into `mockResolvedValue`.

**Reply by pantharshit00** [reply]
Hello @quinnvaughn! Sorry for a late follow up here. 

If this is still relevant, can you please provide a small reproduction in a git repository showing the problem? That way we will be able to try this locally and provide a solution.

**Reply by pantharshit00** [reply]
Hello @quinnvaughn,

I am going to past discussions right now and saw this is still open. Can you please provide the requested details if this is still relevant to you?

**Reply by petja** [reply]
Same problem here!

I guess the issue is on the `jest-mock-extended` end and thus out of scope of the Prisma team. The library isn't aware I have `{ include: {...} }` so it uses the wrong parameter type for `mockResolvedValue`.

This is the scratchpad I used to inspect my problem:
```typescript
// <T extends Prisma.LogbookEntryFindUniqueOrThrowArgs>(args?: Prisma.SelectSubset<T, Prisma.LogbookEntryFindUniqueOrThrowArgs> | undefined) => Prisma.Prisma__LogbookEntryClient<Prisma.LogbookEntryGetPayload<T>, never>
type NotMocked = typeof prisma.logbookEntry.findUniqueOrThrow

// Prisma.Prisma__LogbookEntryClient<LogbookEntry, never>
type NotMockedReturn = ReturnType<NotMocked>

/*
CalledWithMock<Prisma.Prisma__LogbookEntryClient<LogbookEntry, never>, [args?: {
    select?: Prisma.LogbookEntrySelect | null | undefined;
    include?: Prisma.LogbookEntryInclude | null | undefined;
    ... 5 more ...;
    distinct?: Prisma.Enumerable<...> | undefined;
} | undefined]> & (<T extends Prisma.LogbookEntryFindFirstOrThrowArgs>(args?: Prisma.SelectSubset<...> | undefined) => Prisma.Prisma__LogbookEntryClient<...>)
*/
type Mocked = typeof prismaMock.logbookEntry.findFirstOrThrow

// LogbookEntry | Prisma.Prisma__LogbookEntryClient<LogbookEntry, never>
type MockedValue = Parameters<Mocked["mockResolvedValue"]>[0]
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
